picard/ui/mainwindow/__init__.py

```diff
@@ -961,111 +541,114 @@ def _update_cd_lookup_button(self):
     def toggle_rename_files(self, checked):
         config = get_config()
         config.setting['rename_files'] = checked
-        self.update_script_editor_examples()
+        self._update_script_editor_examples()
 
     def toggle_move_files(self, checked):
         config = get_config()
         config.setting['move_files'] = checked
-        self.update_script_editor_examples()
+        self._update_script_editor_examples()
 
     def toggle_tag_saving(self, checked):
         config = get_config()
         config.setting['dont_write_tags'] = not checked
 
-    def get_selected_or_unmatched_files(self):
+    def _get_selected_or_unmatched_files(self):
         if self.selected_objects:
             files = list(iter_files_from_objects(self.selected_objects))
             if files:
                 return files
         return self.tagger.unclustered_files.files
 
     def open_tags_from_filenames(self):
-        files = self.get_selected_or_unmatched_files()
+        files = self._get_selected_or_unmatched_files()
         if files:
             dialog = TagsFromFileNamesDialog(files, self)
             dialog.exec()
 
     def open_collection_in_browser(self):
         self.tagger.collection_lookup()
 
-    def create_menus(self):
+    def _create_menus(self):
+        def add_action(menu, action_id):
+            menu.addAction(self.actions[action_id])
+
         menu = self.menuBar().addMenu(_("&File"))
-        menu.addAction(self.add_directory_action)
-        menu.addAction(self.add_files_action)
+        add_action(menu, MainAction.ADD_DIRECTORY)
+        add_action(menu, MainAction.ADD_FILES)
```

Comment:
Brainstorming here but I'm wondering if it would work (and possibly shorten the code) to somehow iterate through the `MainAction` attributes rather than have separate calls to `add_action()` for each.  I haven't looked at this in detail, but I suspect that it might require additional (conditional) information on each of the attributes.  Or would that actually complicate things again?

Reply:
The thing is that iterate will not give the hierarchy of menus. Nothing much to improve here imho.

Reply:
Okay.  As I said, I was just brainstorming.

Reply:
@rdswift have a look at https://github.com/metabrainz/picard/pull/2450/commits/5429085d7570fa2623c7ae8243a1be5ba2073600
It's based on your idea, what do you think?

Reply:
> Okay. As I said, I was just brainstorming.

No problem, that's welcome, actually it led me to the code you can see in latest commits. It makes menu building much clearer and hides the complexity in a small helper method that replaces previous one.

Reply:
> It's based on your idea, what do you think?

I really like it.  It really simplifies building the menus because it's sort of "what you see is what you get".  Well done!

Reply:
I applied the same medicine (more or less) to itemviews context menu in https://github.com/metabrainz/picard/pull/2450/commits/4ec3b91318fdaa63da816d7de43c250065063e00